lib/Elevate/Leapp.pm

```diff
@@ -308,40 +308,23 @@ Please investigate, resolve then re-run the following command to continue the up
 
 EOS
 
-    my $leapp_json_report = LEAPP_REPORT_JSON;
-    if ( -e $leapp_json_report ) {
-        my $report = eval { Cpanel::JSON::LoadFile($leapp_json_report) } // {};
-
-        my $entries = $report->{entries};
-        if ( ref $entries eq 'ARRAY' ) {
-            foreach my $e (@$entries) {
-                next unless ref $e && $e->{title} =~ qr{Missing.*answer}i;
-
-                $msg .= $e->{summary} if $e->{summary};
-
-                if ( ref $e->{detail} ) {
-                    my $d = $e->{detail};
-
-                    if ( ref $d->{remediations} ) {
-                        foreach my $remed ( $d->{remediations}->@* ) {
-                            next unless $remed->{type} && $remed->{type} eq 'command';
-                            next unless ref $remed->{context};
-                            my @hint = $remed->{context}->@*;
-                            next unless scalar @hint;
-                            $hint[0] = q[/usr/bin/leapp] if $hint[0] && $hint[0] eq 'leapp';
-                            my $cmd = join( ' ', @hint );
-
-                            $msg .= "\n\n";
-                            $msg .= <<"EOS";
+    my $entries = $self->search_report_file_for_inhibitors;
+    if ( ref $entries eq 'ARRAY' ) {
+        foreach my $e (@$entries) {
+            $msg .= "\n$e->{summary}" if $e->{summary};
```

Comment:
There is also a "title" and "hint" value in the hash that can have valuable info.  The Leapp blocker displays this.

Reply:
From the Leapp blocker:
`foreach my $inhibitor (@$inhibitors) {
  69         my $message = $inhibitor->{title} . "\n";
  70         $message .= $inhibitor->{summary} . "\n";
  71         if ( $inhibitor->{hint} ) {
  72             $message .= "Possible resolution: " . $inhibitor->{hint} . "\n";
  73         }
  74         if ( $inhibitor->{command} ) {
  75             $message .= "Consider running:" . "\n" . $inhibitor->{command} . "\n";
  76         }
  77
  78         $self->has_blocker($message);
  79     }`

Reply:
That's true, but it also changes the scope of the case, I would think.